webapp/src/actions.ts

```diff
@@ -15,8 +27,83 @@ export const closeCreateEventModal = () => {
     };
 };
 
-export const autocompleteConnectedUsers = async (input: string) => {
-    const sampleUsers = ['sysadmin', 'user1', 'user2'];
+export const getSiteURL = (state: GlobalState): string => {
+    const config = getConfig(state);
+
+    let basePath = '';
+    if (config && config.SiteURL) {
+        basePath = new URL(config.SiteURL).pathname;
+
+        if (basePath && basePath[basePath.length - 1] === '/') {
+            basePath = basePath.substring(0, basePath.length - 1);
+        }
+    }
+
+    return basePath;
+};
 
-    return sampleUsers.filter(u => u.toLowerCase().includes(input.toLowerCase()));
+export const getPluginServerRoute = (state: GlobalState): string => {
+    const siteURL = getSiteURL(state);
+    return `${siteURL}/plugins/${PluginId}`;
+};
+
+type AutocompleteUser = {
+    mm_id: string
+    mm_username: string
+    mm_display_name: string
 }
+
+export type AutocompleteConnectedUsersResponse = {data?: AutocompleteUser[]; error?: string};
+
+export const autocompleteConnectedUsers = (input: string) => async (dispatch, getState): Promise<AutocompleteConnectedUsersResponse> => {
+    const state = getState();
+    const pluginServerRoute = getPluginServerRoute(state);
+
+    return doFetchWithResponse(`${pluginServerRoute}/autocomplete/users?search=${input}`).
+        then((response) => {
+            return {data: response.data};
+        }).
+        catch((response) => {
+            const error = response.message?.error || 'An error occurred while searching for users.';
+            return {data: [], error};
+        });
+};
+
+export type AutocompleteChannelsResponse = {data?: Channel[]; error?: string};
+
+export const autocompleteUserChannels = (input: string, teamId: string) => async (dispatch, getState): Promise<AutocompleteChannelsResponse> => {
+    const state = getState();
+    const siteURL = getSiteURL(state);
```

Comment:
We could instead store the site url in a constant upon plugin activation, to avoid accessing state when preparing requests

Reply:
Did you put this comment as a reminder or do you want me to address this?